public/src/managers/ServerProvider.ts

```diff
@@ -243,6 +243,67 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         return JSON.parse(JSON.stringify(result.statusCode)) as IStatusCode;
     }
 
+    // getAllGroupLabInfos 
+    public async getAllGroupLabInfos(courseId: number, groupID: number): Promise<IMap<ISubmission>> { 
+        const result = await this.helper.get<ISubmission[]>(
+            ("courses/" + courseId.toString() + "/groups/" + groupID + "/submissions"),
+        );
+        if (result.statusCode === 200 && result.data === undefined) {
```

Comment:
Shouldn't these http codes be predefined constants that are readable?

Reply:
Yes, I agree they should be defined by the framework that is talking to the backend. I'll fix this.

Reply:
It seems like there is no standard definition for HTTP-codes in typescript/javascript, XHtmlRequester which is used has something called ReadyState, but this is not the same thing. 
I looked around and the norm seems to be either use a library which has these definitions or define them ourselves.
Do you have any suggestions on how to solve this?

Reply:
I'm surprised this isn't part of the std.lib or something. But it seems a lot of other people make their own thing, and I found [this one](https://gist.github.com/scokmen/f813c904ef79022e84ab2409574d1b45) which looks pretty reasonable.